inorbit_edge/robot.py

```diff
@@ -1030,14 +1033,21 @@ class RobotSessionPool:
     """Pool of robot sessions that handles connections for many robots in an
     efficient way"""
 
-    def __init__(self, robot_session_factory):
+    def __init__(self, robot_session_factory, robot_config_yaml=None):
         """Creates the pool
         Args:
           - robot_session_factory: factory used to build individual RobotSession
           objects
         """
         self.robot_session_factory = robot_session_factory
         self.robot_sessions = {}
+
+        # If a robots config yaml file was provided, load it
+        try:
```

Comment:
If the `robot_config_yaml` is provided and it fail to load I'd raise the exception instead of defaulting to `{}`

Reply:
@leandropineda done! thanks for your comment. I've added a check and will raise an exception if the yaml can't be loaded.